Source/JavaScriptCore/b3/B3LowerMacros.cpp

```diff
@@ -417,14 +417,187 @@ class LowerMacros {
+                m_value->replaceWithIdentity(result);
+                m_changed = true;
+                break;
+            }
+
+            case VectorNot: {
+                if (!isX86())
+                    break;
+                // x86_64 has no vector bitwise NOT instruction, so we expand vxv.not v into vxv.xor -1, v
+                // here to give B3/Air a chance to optimize out repeated usage of the mask.
+                v128_t mask;
+                mask.u64x2[0] = 0xffffffffffffffff;
+                mask.u64x2[1] = 0xffffffffffffffff;
+                Value* ones = m_insertionSet.insert<Const128Value>(m_index, m_origin, mask);
+                Value* result = m_insertionSet.insert<SIMDValue>(m_index, m_origin, VectorXor, B3::V128, SIMDLane::v128, SIMDSignMode::None, ones, m_value->child(0));
+                m_value->replaceWithIdentity(result);
+                m_changed = true;
+                break;
+            }
+
+            case VectorNeg: {
+                if (!isX86())
+                    break;
+                // x86_64 has no vector negate instruction. For integer vectors, we can replicate negation by
+                // subtracting from zero. For floating-point vectors, we need to toggle the sign using packed
+                // XOR.
+                SIMDValue* value = m_value->as<SIMDValue>();
+                switch (value->simdLane()) {
+                case SIMDLane::i8x16:
+                case SIMDLane::i16x8:
+                case SIMDLane::i32x4:
+                case SIMDLane::i64x2: {
+                    Value* zero = m_insertionSet.insert<Const128Value>(m_index, m_origin, v128_t());
+                    Value* result = m_insertionSet.insert<SIMDValue>(m_index, m_origin, VectorSub, B3::V128, value->simdInfo(), zero, m_value->child(0));
+                    m_value->replaceWithIdentity(result);
+                    m_changed = true;
+                    break;
+                }
+                case SIMDLane::f32x4: {
+                    Value* topBit = m_insertionSet.insert<Const32Value>(m_index, m_origin, -0x80000000);
+                    Value* floatMask = m_insertionSet.insert<Value>(m_index, BitwiseCast, m_origin, topBit);
+                    Value* vectorMask = m_insertionSet.insert<SIMDValue>(m_index, m_origin, VectorSplat, B3::V128, SIMDLane::f32x4, SIMDSignMode::None, floatMask);
+                    Value* result = m_insertionSet.insert<SIMDValue>(m_index, m_origin, VectorXor, B3::V128, SIMDLane::v128, SIMDSignMode::None, m_value->child(0), vectorMask);
+                    m_value->replaceWithIdentity(result);
+                    m_changed = true;
+                    break;
+                }
+                case SIMDLane::f64x2: {
+                    Value* topBit = m_insertionSet.insert<Const64Value>(m_index, m_origin, -0x8000000000000000);
```

Comment:
Probably ULL postfix is necessary for this integer.

Reply:
Updated.